discord/http.py

```diff
@@ -1208,13 +1209,37 @@ def start_forum_thread(
 
         if rate_limit_per_user:
             payload["rate_limit_per_user"] = rate_limit_per_user
+
+        form = [{"name": "payload_json"}]
+        if files:
+            attachments = []
+            for index, file in enumerate(files):
+                attachments.append(
+                    {
+                        "id": index,
+                        "filename": file.filename,
+                        "description": file.description,
+                    }
+                )
+                form.append(
+                    {
+                        "name": f"files[{index}]",
+                        "value": file.fp,
+                        "filename": file.filename,
+                        "content_type": "application/octet-stream",
+                    }
+                )
+            payload["attachments"] = attachments
+
+        form[0]["value"] = utils._to_json(payload)
```

Comment:
It cannot, because the contents of `payload` change after `form` is created to include the attachments.

We could change the inside of the loop to build the json object in-place instead, but other methods building similar structs do not do that.